osf_tests/test_generate_sitemap.py

```diff
@@ -0,0 +1,168 @@
+import os
+
+import pytest
+import mock
+import shutil
+import tempfile
+import xml
+import urlparse
+
+from scripts import generate_sitemap
+from osf_tests.factories import (AuthUserFactory, ProjectFactory, RegistrationFactory, CollectionFactory,
+                                 PreprintFactory, PreprintProviderFactory, EmbargoFactory)
+from website import settings
+
+
+def get_all_sitemap_urls():
+    # Create temporary directory for the sitemaps to be generated
+
+    generate_sitemap.main()
+
+    # Parse the generated XML sitemap file
+    with open(os.path.join(settings.STATIC_FOLDER, 'sitemaps/sitemap_0.xml')) as f:
+        tree = xml.etree.ElementTree.parse(f)
+
+    shutil.rmtree(settings.STATIC_FOLDER)
+
+    # Get all the urls in the sitemap
+    # Note: namespace was defined in the XML file, therefore necessary to include in tag
+    namespace = '{http://www.sitemaps.org/schemas/sitemap/0.9}'
+    urls = [element.text for element in tree.iter(namespace + 'loc')]
+
+    return urls
+
+
+@pytest.mark.django_db
+class TestGenerateSitemap:
+
+    @pytest.fixture(autouse=True)
+    def user_admin_project_public(self):
+        return AuthUserFactory()
+
+    @pytest.fixture(autouse=True)
+    def user_admin_project_private(self):
+        return AuthUserFactory()
+
+    @pytest.fixture(autouse=True)
+    def public_project_for_registration(self, user_admin_project_public):
```

Comment:
I would prefer if all of the fixture were updated to match the naming convention discussed above, but I won't hold this up further.